src/pages/stacks-blockchain/transactions.md

```diff
@@ -0,0 +1,462 @@
+  codeBody: fs.readFileSync('/path/to/contract.clar').toString(),
+  senderKey: 'b244296d5907de9864c0b0d51f98a13c52890be0404e83f273144cd5b9960eed01',
+  network: new StacksTestnet(),
+};
+
+const transaction = await makeContractDeploy(txOptions);
+```
+
+### Smart contract function call
+
+```js
+import { makeContractCall, BufferCV, StacksTestnet } from '@blockstack/stacks-transactions';
+const BigNum = require('bn.js');
+
+const txOptions = {
+  contractAddress: 'SPBMRFRPPGCDE3F384WCJPK8PQJGZ8K9QKK7F59X',
+  contractName: 'contract_name',
+  functionName: 'contract_function',
+  functionArgs: [bufferCVFromString('foo')],
+  senderKey: 'b244296d5907de9864c0b0d51f98a13c52890be0404e83f273144cd5b9960eed01',
+  // attempt to fetch this contracts interface and validate the provided functionArgs
+  validateWithAbi: true,
+  network: new StacksTestnet(),
+};
+
+const transaction = await makeContractCall(txOptions);
+```
+
+### Clarity value types
+
+Building transactions that call functions in deployed clarity contracts requires you to construct valid Clarity Values to pass to the function as arguments. The [Clarity type system](https://github.com/blockstack/stacks-blockchain/blob/master/sip/sip-002-smart-contract-language.md#clarity-type-system) contains the following types:
+
+| Type             | Declaration                           | Description                                                                                                                                                                         |
+| ---------------- | ------------------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Tuple            | `(tuple (key-name-0 key-type-0) ...)` | Typed tuple with named fields                                                                                                                                                       |
+| List             | `(list max-len entry-type)`           | List of maximum length max-len, with entries of type entry-type                                                                                                                     |
+| Response         | `(response ok-type err-type)`         | Object used by public functions to commit their changes or abort. May be returned or used by other functions as well, however, only public functions have the commit/abort behavior |
+| Optional         | `(optional some-type)`                | Option type for objects that can either be (some value) or none                                                                                                                     |
+| Buffer           | `(buff max-len)`                      | Byte buffer or maximum length max-len                                                                                                                                               |
```

Comment:
Did you mean "Byte buffer *with* maximum length `max-len`"?

Should we also add ascii and utf-8 string types to the table?

Reply:
@diwakergupta the SIP doesn't include the string types either: https://github.com/blockstack/stacks-blockchain/blob/develop/sip/sip-002-smart-contract-language.md#types